quantize.py

```diff
@@ -0,0 +1,244 @@
+# This adapts GPTQ's quantization process: https://github.com/IST-DASLab/gptq/
+# E. Frantar et al GPTQ: Accurate Post-training Compression for GPT, arXiv:2210.17323
+# portions copyright by the authors licensed under the Apache License 2.0
+import math
+import sys
+import time
+from pathlib import Path
+from typing import Optional
+import gc
+
+import lightning as L
+import torch
+import tqdm
+
+from lit_llama import LLaMA, Tokenizer
+from lit_llama.utils import EmptyInitOnDevice, save_model_checkpoint
+from lit_llama.quantization import GPTQQuantizer
+
+from datasets import load_dataset
```

Comment:
we would need to add this to requirements.txt I think

Reply:
ok let's do it